src/server/kiwix-serve.cpp

```diff
@@ -384,12 +378,16 @@ int main(int argc, char** argv)
     exit(1);
   }
 
-  std::string host = (server.getIpMode()==kiwix::IpMode::all || server.getIpMode()==kiwix::IpMode::ipv6)
-                     ? "[" + server.getAddress() + "]" : server.getAddress();
-
-  std::string url = "http://" + host + ":" + std::to_string(server.getPort()) + normalizeRootUrl(rootLocation);
-  std::cout << "The Kiwix server is running and can be accessed in the local network at: "
-            << url << std::endl;
+  std::string prefix = "http://";
+  kiwix::IpAddress addresses = server.getAddress();
+  std::string suffix = ":" + std::to_string(server.getPort()) + normalizeRootUrl(rootLocation);
+  std::cout << "The Kiwix server is running and can be accessed in the local network at: ";
+  if(ipMode == kiwix::IpMode::all)
+    std::cout << prefix << addresses.addr << suffix << " and " << prefix << "[" << addresses.addr6 << "]" <<  suffix << std::endl;
+  else if(ipMode == kiwix::IpMode::ipv4)
+    std::cout << prefix << addresses.addr << suffix << std::endl;
+  else if(ipMode == kiwix::IpMode::ipv6)
+    std::cout << prefix << "[" << addresses.addr6 << "]" << suffix << std::endl;
```

Comment:
I strongly think that we should trust what the server sets for the final print message. This an additional reason for line 381 here, because some mode handling is being done in the server as well, as mentioned in my other comment.

What do you think?

Reply:
I think that extra complexity only hurts. Here we only want to print the server addresses in use. Those addresses come from `Server::getAddress()` and we shouldn't rely on anything else to print them.